### DISCUSSION TITLE
Ignoring scripts with .chezmoiignore

### DISCUSSION THREAD

**Discussion by canac** [twpayne/chezmoi#2510] [Q&A] [answered]
Can scripts be ignored with `.chezmoiignore`? My scripts are in `.chezmoiscripts/` and adding them to `.chezmoiignore` doesn't seem to have any effect. Even with a line `*.sh`, all the scripts still run. If `.chezmoiignore` doesn't affect scripts that might be a good thing to document.

**Comment by twpayne** [comment, marked as answer]
Yes. Note that you have to include `.chezmoiscripts/` and remove any attributes from the source path as `.chezmoiignore` works with target paths.

For example, if you have a script in a file called `.chezmoiscripts/run_once_configure-linux.sh` that you don't want to run on non-Linux machines you would put this in your `.chezmoiignore` file:

```
{{ if ne .chezmoi.os "linux" }}
.chezmoiscripts/configure-linux.sh
{{ end }}
```

#2515 adds a test that demonstrates this.

**Reply by canac** [reply]
My problem was the `run_once_before_` prefix. Thanks for the tip!
